src/SixLabors.Fonts/ArrayBuilder{T}.cs

```diff
@@ -153,23 +154,111 @@ private void EnsureCapacity(int min)
                 newCapacity = (uint)min;
             }
 
-            var array = new T[newCapacity];
-
+            T[] array = ArrayPool<T>.Shared.Rent((int)newCapacity);
             if (this.size > 0)
             {
                 Array.Copy(this.data!, array, this.size);
+                ArrayPool<T>.Shared.Return(this.data!);
             }
 
             this.data = array;
         }
     }
 
+    public readonly bool Contains(T item)
+    {
+        for (int i = 0; i < this.size; i++)
+        {
+            if (this[i].Equals(item))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public readonly int IndexOf(T item)
+    {
+        for (int i = 0; i < this.size; i++)
+        {
+            if (this[i].Equals(item))
+            {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    public void Insert(int index, T item)
+    {
+        if (index < 0 || index > this.size)
+        {
+            throw new ArgumentOutOfRangeException(nameof(index));
+        }
+
+        this.Length++;
+        Array.Copy(this.data!, index, this.data!, index + 1, this.size - index - 1);
+        this[index] = item;
+    }
+
+    public bool Remove(T item)
+    {
+        int index = this.IndexOf(item);
+        if (index >= 0)
+        {
+            this.RemoveAt(index);
+            return true;
+        }
+
+        return false;
+    }
+
+    public void RemoveAt(int index)
+    {
+        if (index < 0 || index >= this.size)
+        {
+            throw new ArgumentOutOfRangeException(nameof(index));
+        }
+
+        Array.Copy(this.data!, index + 1, this.data!, index, this.size - index - 1);
+        this.Length--;
+    }
+
+    public void RemoveRange(int start, int count)
+    {
+        if (start < 0 || start >= this.size)
+        {
+            throw new ArgumentOutOfRangeException(nameof(start));
+        }
+
+        if (start + count > this.size)
+        {
+            throw new ArgumentOutOfRangeException(nameof(count));
+        }
+
+        this.Length -= count;
+    }
+
+    public readonly void Sort() => this.data!.AsSpan(0, this.size).Sort();
+
+    public unsafe void Free()
```

Comment:
Why Explicit `Free()` Methods Are a Bad Idea

1. **Breaks .NET Standards**: The `IDisposable` pattern is the established and widely understood approach for resource management in .NET. Introducing a custom `Free()` method deviates from this standard, making the code less intuitive for others.

2. **No `using` Support**: `IDisposable` integrates with the `using` statement, ensuring deterministic cleanup. A `Free()` method requires manual calls, increasing the risk of resource leaks due to missed invocations.

3. **No Safety Net**: Without a fallback like a finalizer, forgotten `Free()` calls can lead to unmanaged resource leaks, especially in edge cases like exceptions or improper usage.

4. **Inconsistent API Design**: Using `Free()` makes the API harder to use and understand, as developers expect `Dispose()` for cleanup tasks.

I would replace `Free()` with the `IDisposable` pattern in types where we can to ensure adherence to .NET conventions, improve usability, and reduce the potential for resource mismanagement. I don't think this type is one of them.

Reply:
> Why Explicit `Free()` Methods Are a Bad Idea
> 
> 1. **Breaks .NET Standards**: The `IDisposable` pattern is the established and widely understood approach for resource management in .NET. Introducing a custom `Free()` method deviates from this standard, making the code less intuitive for others.
> 2. **No `using` Support**: `IDisposable` integrates with the `using` statement, ensuring deterministic cleanup. A `Free()` method requires manual calls, increasing the risk of resource leaks due to missed invocations.
> 3. **No Safety Net**: Without a fallback like a finalizer, forgotten `Free()` calls can lead to unmanaged resource leaks, especially in edge cases like exceptions or improper usage.
> 4. **Inconsistent API Design**: Using `Free()` makes the API harder to use and understand, as developers expect `Dispose()` for cleanup tasks.
> 
> I would replace `Free()` with the `IDisposable` pattern in types where we can to ensure adherence to .NET conventions, improve usability, and reduce the potential for resource mismanagement. I don't think this type is one of them.

The reason here:
1. It has complex reference so I can't transfer it into struct.
2. this object is little object, so it will collect by GC gen 0. That is way more faster than GC gen 1 for large object. 
3. It only create 1 times when generate glyph layout.

Reply:
I have replaced the `Free` to `Dispose`, this part should close.